src/main/java/ca/nrc/cadc/beacon/web/resources/PackageServerResource.java

```diff
@@ -0,0 +1,207 @@
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *
+ ************************************************************************
+ */
+
+package ca.nrc.cadc.beacon.web.resources;
+
+import ca.nrc.cadc.beacon.web.restlet.JSONRepresentation;
+
+import ca.nrc.cadc.reg.Standards;
+import ca.nrc.cadc.vos.client.ClientTransfer;
+import ca.nrc.cadc.vos.client.VOSpaceClient;
+
+import ca.nrc.cadc.vos.Direction;
+import ca.nrc.cadc.vos.Protocol;
+import ca.nrc.cadc.vos.Transfer;
+import ca.nrc.cadc.vos.VOS;
+import ca.nrc.cadc.vos.View;
+
+import java.net.URI;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import org.apache.log4j.Logger;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.json.JSONWriter;
+import org.restlet.ext.json.JsonRepresentation;
+import org.restlet.representation.Representation;
+import org.restlet.resource.Get;
+import org.restlet.resource.Post;
+
+
+public class PackageServerResource extends StorageItemServerResource {
+    private static Logger log = Logger.getLogger(PackageServerResource.class);
+
+    /**
+     * Empty constructor needed for Restlet to manage it.
+     */
+    public PackageServerResource() {
+    }
+
+    PackageServerResource(final VOSpaceClient voSpaceClient) {
+        super(voSpaceClient);
+    }
+
+    @Get("json")
+    public Representation notSupported() throws Exception {
+        return new JSONRepresentation() {
+            @Override
+            public void write(final JSONWriter jsonWriter)
+                throws JSONException {
+                jsonWriter.object()
+                    .key("msg").value("GET not supported.")
+                    .endObject();
+            }
+        };
+    }
+
+    @Post("json")
+    public Representation getPackage(final JsonRepresentation payload) throws Exception {
+        final JSONObject jsonObject = payload.getJsonObject();
+        log.debug("getPackage input: " + jsonObject);
+
+        List<URI> targetList = new ArrayList<>();
+        final Set<String> keySet = jsonObject.keySet();
+
+        String responseFormat;
+        if (keySet.contains("responseformat")) {
+            responseFormat = jsonObject.getString("responseformat");
+        } else {
+            // default response format
+            responseFormat = "application/zip";
+        }
+
+        if (!keySet.contains("target")) {
+            return new JSONRepresentation() {
+                @Override
+                public void write(final JSONWriter jsonWriter)
+                    throws JSONException {
+                    jsonWriter.object()
+                        .key("msg").value("no targets found.")
+                        .endObject();
+                }
+            };
+        } else {
+            final String targetStr = (String) jsonObject.get("target");
```

Comment:
Is this an actual JSON Array?  You can use `jsonObject.getJSONArray("target")` to get the proper array rather than splitting commas.

Reply:
It's a proper JSON array now.